src/Components/Main/BookLoader/BookList/BookCard/BookCard.jsx

```diff
@@ -0,0 +1,101 @@
+import { useContext, useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+
+import style from "./BookCard.module.scss";
+import defaultBookImg from "../../../../../assets/default_book_cover.jpg";
+
+import BookModal from "../../../BookModal/BookModal";
+
+const BookCard = ({ rawBook }) => {
+  const [showModal, setShowModal] = useState(false);
+
+  const book = {
+    title: rawBook.title,
+    infoLink: rawBook.infoLink,
+    averageRating: rawBook.averageRating ? String(rawBook.averageRating) : "",
+    ratingsCount: rawBook.ratingsCount ? String(rawBook.ratingsCount) : "",
+  };
+  // book image
+  let bookImg = defaultBookImg;
+  if (rawBook.imageLinks) {
+    if (rawBook.imageLinks.small) {
+      bookImg = rawBook.imageLinks.small;
+    }
+    if (rawBook.imageLinks.thumbnail) {
+      bookImg = rawBook.imageLinks.thumbnail;
+    }
+  }
+  book.bookImg = bookImg;
+
+  // book desc
+  book.desc = rawBook.description ? rawBook.description : "";
+
+  // book.authors is Arr of Str ->
+  book.authors = rawBook.authors ? rawBook.authors.join(" & ") : "";
+
+  // industryIdentifiers - ISBN
+
+  if (rawBook.industryIdentifiers && rawBook.industryIdentifiers.length === 2) {
```

Comment:
add a utils folder for pure js functions, create a cleanBookData function that will deal with missing properties etc, return all the data you need